pkg/checks/checks.go

```diff
@@ -0,0 +1,169 @@
+package checks
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appclient "k8s.io/client-go/kubernetes/typed/apps/v1"
+)
+
+// WaitForHTTPStatusOk poll until URL returns 200
+func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urltocheck string) error {
+	cli := &http.Client{
+		Transport: transport,
+		Timeout:   10 * time.Second,
+	}
+
+	req, err := http.NewRequest("GET", urltocheck, nil)
+	if err != nil {
+		return err
+	}
+
+	for {
+		resp, err := cli.Do(req)
+		if err, ok := err.(*url.Error); ok && (err.Timeout() || err.Err == io.EOF || err.Err == io.ErrUnexpectedEOF) {
+			time.Sleep(time.Second)
+			continue
+		}
+
+		if err == io.EOF {
```

Comment:
the implementation changed upstream while your PR was in progress (see the `return wait.PollInfinite`...) and you'll need to merge my changes and yours.  The end result should include at least the `wait.PollInfinite`, the `syscall.ENETUNREACH` bit and your new `err == io.EOF` bit.